Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -59,23 +59,22 @@ class TrivialBookmarkStorer: BookmarkStorer {
         var success: [GUID] = []
         var failed: [GUID: String] = [:]
 
-        func onUpload(result: POSTResult) -> DeferredTimestamp {
-            modified = result.modified
+        func onUpload(result: POSTResult, lastModified: Timestamp?) -> DeferredTimestamp {
             success.appendContentsOf(result.success)
             result.failed.forEach { guid, message in
                 failed[guid] = message
             }
 
-            log.debug("Uploaded records got timestamp \(modified).")
-            local.setModifiedTime(modified, guids: result.success)
-            return deferMaybe(result.modified)
+            log.debug("Uploaded records got timestamp \(lastModified).")
+            local.setModifiedTime(lastModified ?? modified, guids: result.success)
```

Comment:
AFAICS `modified` is no longer set, so it will always be `0`. Is that what you want?